.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,58 @@
+name: Run-Tests
```

Comment:
The file name is `unit_test.yaml`. But this does three types of tests.
Maybe let's rename this to something better describing this complexity and `unit_test.yaml` have the `unit-test` job?

Reply:
the file name is bad. It was when my first PoC had 4 files. Ended with one and forgot to rename it. I would rename the file to `pr_tests.yml`. How about that?